types/field_type.go

```diff
@@ -133,7 +133,11 @@ func DefaultParamTypeForValue(value interface{}, tp *FieldType) {
 		tp.Flen = UnspecifiedLength
 		tp.Decimal = UnspecifiedLength
 	default:
+		flen := tp.Flen
 		DefaultTypeForValue(value, tp)
+		if tp.Tp == mysql.TypeVarString {
+			tp.Flen = flen
```

Comment:
Seems saved `flen` is always `UnspecifiedLength`, can we directly assign this to `tp.Flen` and remove line 136?